packages/components/components/elvis-context-menu/src/react/elvia-context-menu.tsx

```diff
@@ -17,98 +30,115 @@ export const ContextMenu: React.FC<ContextMenuProps> = ({
   trigger,
   inlineStyle,
   display = 'inline-block',
+  anchorPosition,
   className,
   webcomponent,
   ...rest
 }) => {
-  const popoverRef = useRef<HTMLDivElement>(null);
   const triggerRef = useRef<HTMLDivElement>(null);
-  const { trapFocus, releaseFocusTrap } = useFocusTrap();
-  const [prevFocusedElement, setPrevFocusedElement] = useState<HTMLElement>();
-  const [fadeOut, setFadeOut] = useState(false);
+  const [isOpen, setIsOpen] = useState(isShowing);
 
   useSlot('trigger', webcomponent, { ref: triggerRef });
 
-  const { isShowing: isOverlayShowing = false, setIsShowing: setIsOverlayShowing } = useConnectedOverlay(
-    triggerRef,
-    popoverRef,
-    {
-      horizontalPosition: mapPositionToHorizontalPosition(horizontalPosition),
-      verticalPosition: verticalPosition,
-      alignWidths: false,
+  const updateOpenState = (newState: boolean) => {
+    setIsOpen(newState);
+    if (newState) {
+      handleOnOpen();
+    } else {
+      handleOnClose();
+    }
+  };
+
+  const { refs, floatingStyles, context } = useFloating({
+    placement: mapPosition(verticalPosition, horizontalPosition),
+    open: isOpen,
+    onOpenChange: updateOpenState,
+    elements: {
+      reference: triggerRef.current,
     },
-  );
+    middleware: [offset({ mainAxis: 8 }), flip(), shift()],
+    whileElementsMounted: autoUpdate,
+    transform: false,
+  });
+
+  // This hook enables the floating element to be positioned based on the anchorPosition prop
+  const clientPoint = useClientPoint(context, {
+    enabled: !!anchorPosition,
+    x: anchorPosition?.left,
+    y: (anchorPosition?.top ?? 0) - window.scrollY, // Adjust for scroll position so the context menu is not positioned "fixed"
+  });
+
+  const dismiss = useDismiss(context);
+
+  const { getReferenceProps, getFloatingProps } = useInteractions([dismiss, clientPoint]);
+
+  const { isMounted, styles: transitionStyles } = useTransitionStyles(context, {
+    duration: 300,
+    common: {
+      transitionTimingFunction: 'ease',
+      zIndex: 99999,
+      maxWidth: 'calc(100% - 16px)',
+    },
+    close: {
+      opacity: 0,
+      transform: 'scale(.9)',
+    },
+    open: {
+      opacity: 1,
+      transform: 'scale(1)',
+    },
```

Comment:
It would be fun if these transition styles included a `transformOrigin` property, making the floating element shrink toward its trigger when it closes 🧑‍🎨

Reply:
That's cool! And the `useFloating` hook does provide the final `placement` of the floating element (e.g. `top-start`) so that I can use that to set the `transformOrigin` to the top or bottom depending on where it is placed 💡